internal/provider/vaultradar/resource_radar_source_github_enterprise.go

```diff
@@ -0,0 +1,235 @@
+	}
+
+	res, err := clients.OnboardRadarSource(ctx, r.client, projectID, radar_service.OnboardDataSourceBody{
+		Type:          "github_enterprise",
+		Name:          plan.GitHubOrganization.ValueString(),
+		ConnectionURL: plan.DomainName.ValueString(),
+		Token:         plan.Token.ValueString(),
+	})
+	if err != nil {
+		resp.Diagnostics.AddError("Error creating Radar source", err.Error())
+		return
+	}
+
+	plan.ID = types.StringValue(res.GetPayload().ID)
+	plan.ProjectID = types.StringValue(projectID)
+	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
+
+	tflog.Trace(ctx, "Created Radar resource")
+}
+
+func (r *sourceGitHubEnterpriseResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state gitHubEnterpriseSource
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	projectID := r.client.Config.ProjectID
+	if !state.ProjectID.IsUnknown() {
+		projectID = state.ProjectID.ValueString()
+	}
+
+	res, err := clients.GetRadarSource(ctx, r.client, projectID, state.ID.ValueString())
+	if err != nil {
+		if clients.IsResponseCodeNotFound(err) {
+			// Resource is no longer on the server.
+			tflog.Info(ctx, "Radar source not found, removing from state.")
+			resp.State.RemoveResource(ctx)
+			return
+		}
+		resp.Diagnostics.AddError("Unable to get Radar source", err.Error())
+		return
+	}
+
+	// Resource is marked as deleted on the server.
+	if res.GetPayload().Deleted {
+		tflog.Info(ctx, "Radar source marked for deletion, removing from state.")
+		resp.State.RemoveResource(ctx)
+		return
+	}
+
+	// The only other state that could change related to this resource is the token, and for obvious reasons we don't
+	// return that in the read response. So we don't need to update the state here.
+	tflog.Trace(ctx, "Read Radar resource")
+}
+
+func (r *sourceGitHubEnterpriseResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var state gitHubEnterpriseSource
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	projectID := r.client.Config.ProjectID
+	if !state.ProjectID.IsUnknown() {
+		projectID = state.ProjectID.ValueString()
+	}
+
+	// Assert resource still exists.
+	res, err := clients.GetRadarSource(ctx, r.client, projectID, state.ID.ValueString())
+	if err != nil {
+		if clients.IsResponseCodeNotFound(err) {
+			// Resource is no longer on the server.
+			tflog.Info(ctx, "Radar source not found, removing from state.")
+			resp.State.RemoveResource(ctx)
+			return
+		}
+		resp.Diagnostics.AddError("Unable to get Radar source", err.Error())
+		return
+	}
+
+	// Resource is already marked as being deleted on the server.
+	if res.GetPayload().Deleted {
+		tflog.Info(ctx, "Radar source marked for deletion, removing from state.")
+		resp.State.RemoveResource(ctx)
+		return
+	}
```

Comment:
Just curious, should we wait for the Vault Radar datasource to be deleted in this case?

Reply:
Makes sense. I will refactor this to wait on the delete to complete if it was already initialize...

This also makes sense in the case that delete was called, inited, but didn't complete within the timeout of 10m, then they can run this again, picking up that its already in a delete in progress.